modules/gapi/include/opencv2/gapi/video.hpp

```diff
@@ -62,6 +62,60 @@ G_TYPED_KERNEL(GCalcOptFlowLKForPyr,
         return std::make_tuple(empty_array_desc(), empty_array_desc(), empty_array_desc());
     }
 };
+
+enum BackgroundSubtractorType
+{
+    TYPE_BS_MOG2,
+    TYPE_BS_KNN
+};
+
+/** @brief Structure for initialization of the Background Subtractor operation's parameters.*/
+
+struct BackgroundSubtractorParams
+{
+    //!< Type of the Background Subtractor operation.
+    BackgroundSubtractorType operation = TYPE_BS_MOG2;
+    //!< Length of the history.
+    int history = 500;
+    //!< Threshold on the squared distance.
```

Comment:
Is it different depending on the type of subtractor? It is said in the constructor

Reply:
Changed